src/quest_manager/models.py

```diff
@@ -271,6 +279,13 @@ def not_expired(self):
 
     def visible(self):
         return self.filter(visible_to_students=True)
+    
+    def active_or_no_campaign(self):
+        """With self as an argument, returns a filtered queryset 
+        containing only quests in active campaigns or quests without campaigns.
+        """
+        active_campaigns = Category.objects.filter(active=True)
+        return (self.filter(Q(campaign__in=active_campaigns) | Q(campaign=None)))
```

Comment:
instead of getting all the active campaigns and checking if it's in the list, could you do something like this instead?  You can traverse foreign key fields using double underscores
`campaign__active=True`

Though, you still might need the Q object for the "OR"

ACTUALY!  Maybe you could exclude instead of filter, so it would just be: `self.exclude(campaign__active=False`  one line and much nicer.  However, it assumes that None will NOT be counted as False, which is probably not the case.

Reply:
self.exclude(campaign__active=False) works perfectly with only one line, thank you. The control quest without a campaign returns active in testing all the same.

Reply:
Awesome.